src/output-eve-null.c

```diff
@@ -0,0 +1,85 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Jeff Lucovsky <jlucovsky@oisf.net>
+ *
+ * File-like output for logging: null/discard device
+ */
+
+#include "suricata-common.h" /* errno.h, string.h, etc. */
+
+#include "output.h" /* DEFAULT_LOG_* */
+#include "output-eve-null.h"
+
+#ifdef OS_WIN32
```

Comment:
why is this still needed? Should be explained in the git message I think

Reply:
The dynamic loading library component isn't available so `HAVE_PLUGIN` is false. When false, the symbol `SCRegisterEveFileType` is not available.

I'll update the git commit and resubmit.

Reply:
Is this the desired state though? We're not loading a real plugin, are we? So perhaps `SCRegisterEveFileType` should just be available?

Reply:
Quick fix, just remove the registration outside of these guards.  Better fix is a bit of refactoring to make registerable file types less like plugins.  Since this is being targetted for 8, maybe it should hold off until I get to that.  Its not super quick as if you do the simple move the functions around you run into some header and include order issues that I'd like to clean up properly as well.

Reply:
@jasonish Would it make sense to include my commits with your refactoring PR?